server/service/src/requisition/program_settings/customer_program_settings/prepare.rs

```diff
@@ -50,73 +86,121 @@ pub(super) fn prepare_customer_program_settings(
 
     let order_types = ProgramRequisitionOrderTypeRowRepository::new(&ctx.connection)
         .find_many_by_program_requisition_settings_ids(&program_requisition_settings_ids)?;
+
     // Periods (matching settings program_schedule_ids)
-    let program_schedule_ids = settings
+    let program_schedule_ids: Vec<&str> = settings
         .iter()
         .map(|s| s.program_settings_row.period_schedule_id.as_str())
         .collect();
 
     let periods = PeriodRowRepository::new(&ctx.connection)
         .find_many_by_program_schedule_ids(program_schedule_ids)?;
 
-    let period_ids = periods.iter().map(|p| p.id.clone()).collect();
+    let period_ids: Vec<String> = periods.iter().map(|p| p.id.clone()).collect();
 
     // Requisitions in Period (for all periods and store)
     let filter = RequisitionsInPeriodFilter::new()
-        .store_id(equal_to_store_id)
+        .other_party_id(EqualFilter::equal_to(customer_name_id))
         .program_id(EqualFilter::equal_any(program_ids.clone()))
         .period_id(EqualFilter::equal_any(period_ids))
         .r#type(RequisitionType::Response.equal_to());
 
     let requisitions_in_periods =
         RequisitionsInPeriodRepository::new(&ctx.connection).query(filter)?;
 
-    let filter = ProgramCustomerFilter::new().program_id(EqualFilter::equal_any(program_ids));
-    let program_customers =
-        ProgramCustomerRepository::new(&ctx.connection).query(store_id, filter)?;
+    Ok(CustomerProgramRequisitionSetting {
+        customer_name_id: customer_name_id.to_owned(),
+        master_lists: settings
+            .iter()
+            .map(|setting| {
+                let order_types_mapped = order_types
+                    .iter()
+                    // only map order types to their respective program requisition settings
+                    .filter(|order_type| {
+                        order_type
+                            .program_requisition_settings_id
+                            .eq(&setting.program_settings_row.id)
+                    })
+                    .map(|order_type| {
+                        // only map periods (and their availability) to their respective order types
+                        map_period_rows_and_requisitions_to_order_type(
+                            order_type.clone(),
+                            periods.clone(),
+                            setting,
+                            requisitions_in_periods.clone(),
+                        )
+                    })
+                    // filter out order types which do not have available periods
+                    .filter(|order_type_with_periods| {
+                        order_type_with_periods.available_periods.len() > 0
+                    })
```

Comment:
Filter order types to those with periods available here rather than filtering on front end to be consumer focussed.